multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -60,3 +122,179 @@ func (r *ClusterSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&multiclusterv1alpha1.ClusterSet{}).
 		Complete(r)
 }
+
+func (r *ClusterSetReconciler) validateLocalClusterClaim(clusterSet *multiclusterv1alpha1.ClusterSet) error {
+	// read the cluster claim in the cluster
+
+	configNamespace := clusterSet.GetNamespace()
+
+	clusterClaimList := &multiclusterv1alpha1.ClusterClaimList{}
+	r.Log.Info("Validating cluster claim in", "namespace", configNamespace)
+	if err := r.List(context.TODO(), clusterClaimList, client.InNamespace(configNamespace)); err != nil {
+		return err
+	}
+	if len(clusterClaimList.Items) == 0 {
+		return fmt.Errorf("cluster claim not configured for the cluster")
+	}
+
+	var clusterSetClaimID string
+	wellKnownClusterSetClaimIDExist := false
+	var clusterClaimID string
+	wellKnownClusterClaimIDExist := false
+	for _, clusterClaim := range clusterClaimList.Items {
```

Comment:
I didn't get this check, should this kind of logic being in clusterclaim_controller.go?

Reply:
To capture the offline conversation:
The idea is that we allow the customer to create ClusterSet CRD before creating the ClusterClaim CRDs and the reconciler will keep retrying processing the ClusterSet CRD until a valid ClusterClaim CRD is created.

Another way is: maybe move this validation to ClusterClaim controller, but since a valid CRD needs to process the ClusterSet, the current way maybe better.

A third way maybe to move this into a validator which may prevent ClusterSet CRD from being created, this may require customer to retry configuring the ClusterSet CRD. I think the current approach maybe better, so the customer can see the errors and fix it.

Reply:
any thoughts on this?